internal/admin/handle_export_importers.go

```diff
@@ -0,0 +1,157 @@
+			ErrorPage(c, err.Error())
+			return
+		}
+
+		ctx := c.Request.Context()
+		m := TemplateMap{}
+
+		db := database.New(s.env.Database())
+		model := new(model.ExportImport)
+
+		idRaw := c.Param("id")
+		if idRaw != "" && idRaw != "0" {
+			id, err := strconv.ParseInt(idRaw, 10, 64)
+			if err != nil {
+				ErrorPage(c, "failed to to parse `id` param.")
+				return
+			}
+
+			model, err = db.GetConfig(ctx, id)
+			if err != nil {
+				ErrorPage(c, fmt.Sprintf("failed to load export importer config: %s", err))
+				return
+			}
+		}
+
+		if err := form.BuildExportImporterModel(model); err != nil {
+			ErrorPage(c, fmt.Sprintf("failed to build export importer config: %s", err))
+			return
+		}
+
+		fn := db.AddConfig
+		if model.ID != 0 {
+			fn = db.UpdateConfig
+		}
+
+		if err := fn(ctx, model); err != nil {
+			ErrorPage(c, fmt.Sprintf("failed to write export importer config: %s", err))
+			return
+		}
+
+		m.AddSuccess("Successfully updated export importer config!")
+		m["model"] = model
+		c.HTML(http.StatusOK, "export-importer", m)
+		c.Abort()
+	}
+}
+
+// HandleExportImportersShow handles the create/update actions for export
+// importers.
+func (s *Server) HandleExportImportersShow() func(c *gin.Context) {
+	return func(c *gin.Context) {
+		ctx := c.Request.Context()
+
+		db := database.New(s.env.Database())
+		model := new(model.ExportImport)
+
+		if idRaw := c.Param("id"); idRaw != "" && idRaw != "0" {
+			id, err := strconv.ParseInt(idRaw, 10, 64)
+			if err != nil {
+				ErrorPage(c, fmt.Sprintf("Failed to parse `id` param: %s", err))
+				return
+			}
+
+			model, err = db.GetConfig(ctx, id)
+			if err != nil {
+				ErrorPage(c, fmt.Sprintf("Failed to load export importer config: %s", err))
+				return
+			}
+		}
+
+		m := make(TemplateMap)
+		m["model"] = model
+		c.HTML(http.StatusOK, "export-importer", m)
+		c.Abort()
+	}
+}
+
+type exportImporterFormData struct {
+	IndexFile  string `form:"index-file"`
+	ExportRoot string `form:"export-root"`
+	Region     string `form:"region"`
+
+	// FromDate and FromTime are combined into FromTimestamp.
+	FromDate string `form:"from-date"`
+	FromTime string `form:"from-time"`
+
+	// ThruDate and ThruTime are combined into ThruTimestamp.
+	ThruDate string `form:"thru-date"`
+	ThruTime string `form:"thru-time"`
+}
+
+// BuildExportImporterModel populates and mutates the given model with form
+// data. It overwrites any form data that's present.
+func (f *exportImporterFormData) BuildExportImporterModel(c *model.ExportImport) error {
+	from, err := CombineDateAndTime(f.FromDate, f.FromTime)
+	if err != nil {
+		return err
+	}
+	thru, err := CombineDateAndTime(f.ThruDate, f.ThruTime)
+	if err != nil {
+		return err
+	}
+
+	if val := f.IndexFile; val != "" {
+		c.IndexFile = val
+	}
+
+	if val := f.ExportRoot; val != "" {
+		c.ExportRoot = val
+	}
+
+	if val := f.Region; val != "" {
+		c.Region = val
+	}
+
+	if !from.IsZero() {
+		c.From = from
+	}
+
+	if !thru.IsZero() {
+		c.Thru = &thru
+	}
```

Comment:
Please beware that if at any point a time.Time value was created by unmarshaling Seconds and Nanosecond respectively and it got the zero value, these checks will fail as per https://play.golang.org/p/ma-1o9J4ynh or inlined  below
```go
package main

import (
	"encoding/json"
	"time"
)

func main() {
	type recv struct {
		Sec int64 `json:"sec"`
		Ns  int64 `json:"ns"`
	}
	rc := new(recv)
	if err := json.Unmarshal([]byte(`{}`), rc); err != nil {
		panic(err)
	}
	t := time.Unix(rc.Sec, rc.Ns)
	println("isZero", t.IsZero(), "unix <= 0", t.UnixNano() <= 0)
}
```
which prints `isZero false unix <= 0 true`

For that I'd highly recommend using `(time.Time).UnixNano() > 0`

Reply:
This PR is just moving existing code around. To reduce the size of a diff and to reduce the chance of introducing a bug, I'm only moving code, no refactors or other changes in this PR.